test/e2e/antreapolicy_test.go

```diff
@@ -2761,6 +2764,40 @@ func TestAntreaPolicyStatus(t *testing.T) {
 	assert.NoError(t, err, "Antrea ClusterNetworkPolicy failed to reach expected status")
 }
 
+// waitForANPRealized waits untils an ANP is realized and returns, or times out. A policy is
+// considered realized when its Status has been updated with DesiredNodesRealized > 0 and
+// CurrentNodesRealized == DesiredNodesRealized.
+func (data *TestData) waitForANPRealized(t *testing.T, namespace string, name string) error {
+	t.Logf("Waiting for ANP '%s/%s' to be realized", namespace, name)
+	if err := wait.Poll(100*time.Millisecond, policyRealizedTimeout, func() (bool, error) {
+		anp, err := data.crdClient.CrdV1alpha1().NetworkPolicies(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return anp.Status.DesiredNodesRealized > 0 && anp.Status.CurrentNodesRealized == anp.Status.DesiredNodesRealized, nil
```

Comment:
Sorry for the late review. If we want this method to be more generic, should it check if the `Generation` in `Status` equals to the one in `Spec` and the `Phase` is `Realized`? Otherwise a policy selecting no appliedto Pods will always time out and an updated NetworkPolicy will be considered realized early than it is.

Reply:
No worries, I made the suggested changes in https://github.com/antrea-io/antrea/pull/2277